src/test/java/com/znsio/teswiz/screen/android/vodqa/VodqaScreenAndroid.java

```diff
@@ -0,0 +1,55 @@
+package com.znsio.teswiz.screen.android.vodqa;
+
+import com.znsio.teswiz.runner.Driver;
+import com.znsio.teswiz.runner.Visual;
+import com.znsio.teswiz.screen.android.theapp.AppLaunchScreenAndroid;
+import com.znsio.teswiz.screen.vodqa.VodqaScreen;
+import org.apache.log4j.Logger;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+
+import static com.znsio.teswiz.tools.Wait.waitFor;
+
+public class VodqaScreenAndroid extends VodqaScreen {
+    private static final String SCREEN_NAME = VodqaScreenAndroid.class.getSimpleName();
+    private static final Logger LOGGER = Logger.getLogger(SCREEN_NAME);
+    private final Driver driver;
+    private final Visual visually;
+    private static final By byLoginButtonXpath = By.xpath("//android.view.ViewGroup[@content-desc=\"login\"]");
+    private static final By byNativeViewXpath = By.xpath("//android.widget.TextView[@content-desc=\"chainedView\"]");
+    private static final String byPageHeaderXpath = "//android.widget.TextView[@text='%s']";
+
+    public VodqaScreenAndroid(Driver driver, Visual visually) {
+        this.driver = driver;
+        this.visually = visually;
+    }
+
+    @Override
+    public VodqaScreen login() {
+        visually.checkWindow(SCREEN_NAME, "Login Page");
+        driver.findElement(byLoginButtonXpath).click();
+        return this;
+    }
+
+    @Override
+    public VodqaScreen tapInTheMiddle() {
+        driver.waitTillElementIsVisible(byNativeViewXpath);
+        visually.checkWindow(SCREEN_NAME, "Sample List page");
+        driver.tapOnMiddleOfScreen();
+        return this;
+    }
+
+    @Override
+    public Boolean verifyLoginPage() {
+        driver.waitTillElementIsPresent(byLoginButtonXpath);
```

Comment:
this line contradicts with the next line, if you are waiting till element is present then there is no need for checking isDisplayed(). If still this needs to be checked, then it only make sense to keep consistency and check for all the elements not just one.

What I propose is you can simply add this driver.waitTillElementIsPresent(byLoginButtonXpath); in login method itself before clicking on login button. Also remove the call from BL, method from Screen, and method definition from ScreenAndroid